Include AppStream metadata info into source distribution

pemensik opened the pull request:
Taken from Flathub package at:
https://github.com/flathub/org.openttd.OpenTTD

With few changes to add more url links and use SPDX license identifier.

## Motivation / Problem
Openttd is a nice graphical application, but lacks AppStream metainfo file in source archive. I think it would be nice to include it directly with the game, so GUI frontends like GNOME Software can show nice information about the game.

<!--
Describe here shortly
* For bug fixes:
    * What problem does this solve?
    * If there is already an issue, link the issue, otherwise describe the problem here.
* For features or gameplay changes:
    * What was the motivation to develop this feature?
    * Does this address any problem with the gameplay or interface?
    * Which group of players do you think would enjoy this feature?
-->


## Description

<!--
Describe here shortly
* For bug fixes:
    * How is the problem solved?
* For features or gameplay changes:
    * What does this feature do?
    * How does it improve/solve the situation described under 'motivation'.
-->


## Limitations

<!--
Describe here
* Is the problem solved in all scenarios?
* Is this feature complete? Are there things that could be added in the future?
* Are there things that are intentionally left out?
* Do you know of a bug or corner case that does not work?
-->


## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

pemensik committed:
Include AppStream metadata info into source distribution

Taken from Flathub package at:
https://github.com/flathub/org.openttd.OpenTTD

With few changes to add more url links and use SPDX license identifier.
Fix license to GPL-2.0-only.

TrueBrain commented on media/org.openttd.OpenTTD.metainfo.xml:
```diff
@@ -0,0 +1,59 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--Copyright 2018-2020 Flathub maintainers-->
```
They cannot be the copyright holder if it is included in our source-tree. So you will have to check with them if they are willing to transfer the copyright to us.

pemensik replied:
The file is under CC0-1.0 license, so this should be just about having authors specified. But I have created issue for that, so they can react any way.

TrueBrain replied:
Copyright and license are two different things :) They can license it under CC0, but they still hold the copyright :) A bit silly for a file like this, ofc, but they added the copyright, not me :P Tnx for following up on it!

ghisvail replied:
> They can license it under CC0, but they still hold the copyright

"The person who associated a work with this deed has dedicated the work to the public domain by **waiving all of his or her rights to the work worldwide under copyright law**, including all related and neighboring rights, to the extent allowed by law." [CC0-1.0](https://creativecommons.org/publicdomain/zero/1.0/deed)

ghisvail replied:
> A bit silly for a file like this

Your judgement call put aside, the copyright header was more of an authorship statement than anything. Copyright is waived as per the quote of the CC0 above. Feel free to remove it.

TrueBrain commented on media/org.openttd.OpenTTD.metainfo.xml:
```diff
@@ -0,0 +1,59 @@
+    <release version="1.10.2" date="2020-06-01"/>
+    <release version="1.10.1" date="2020-04-13"/>
+    <release version="1.10.0" date="2020-04-01"/>
+  </releases>
```
This `<release>` blob is impractical for us to maintain. Please remove it.

glx22 replied:
It might be possible to auto-fill it by parsing changelog when configuring/building

TrueBrain replied:
Please don't. It will break. We will not find out :) We have been around this bush a few times already ;)

ghisvail replied:
These are automatically added by the Flatpak update bot. It's okay to drop them and only maintain the latest release entry.

pemensik replied:
Understood, having extra release tags is impractical. But what if we had a tool to generate it instead? Just like shell script:

```
#!/bin/sh
# Generate metainfo releases from git tags
#

# <release version="1.10.1" date="2020-04-13"/>
#
VERSIONS=${1:-10}

get_date() {
	git log -1 --pretty=format:%as $@
}

get_tags() {
	git tag -l | grep -Ev -- '-(RC|beta).*' | sort -r --version-sort
}

print_release() {
	local VER="$1"
	DATE=$(get_date $VER)
	echo "<release version=\"$VER\" date=\"$DATE\"/>"
}

if [ -n "$VERSIONS" ] && [ "$VERSIONS" != all ]; then
	get_tags | head -${VERSIONS} | while read VER; do print_release $VER; done
else
	get_tags | while read VER; do print_release $VER; done
fi
```

Would that be acceptable instead?

TrueBrain replied:
The issue here too is that we have to maintain these things. And we are having a hard time as it is to maintain our current release version etc. We often even forget to the the simplest thing upon release, causing issues like 13.2 / 13.3 / 13.4.

Especially for files that won't show up in a short period of time after release that they are borked, it is very annoying to throw any automation against it.

But maybe more of a question: what value does it add to see the version history in a file like this? If it is just to help GNOME etc to show OpenTTD, why would it want to know the history? (honest question, to be clear)

TrueBrain replied:
Btw, if you really need an entry in the releases list, you could do something similar as with `openttd.desktop` (which needs the actual binary name), and have a single entry which is replaced with the current version the game is building for.

This would be a bit of work, as you need to do something similar for `openttd.desktop` in `media/CMakeLists.txt`, and update `cmake/scripts/Desktop.cmake` to also be able to replace the version. But that way you could have the latest version in the list there, in an automated way, that doesn't require any real maintenance from our side.

But if we can just not have the list, that would work too. Fully depends on what it is actually used for :)

pemensik replied:
Yes, at least one release record is mandatory for strictly validated metainfo file. Problem is we need also the release date, not only current version. Having only single record would be a bit poor choice, but better than nothing at all. Ideally the release tag should contain short summary about what is new in that version.

TrueBrain replied:
Short summary we don't have. Like, literally, we don't have it :P We make a short news post poking fun of things, but that is about as far as it gets :D

There is `REV_ISODATE`, which is the date of the current build you are running. For a release that will be the date of the release.

TrueBrain commented on media/org.openttd.OpenTTD.metainfo.xml:
```diff
@@ -0,0 +1,59 @@
+    <release version="1.10.1" date="2020-04-13"/>
+    <release version="1.10.0" date="2020-04-01"/>
+  </releases>
+  <content_rating type="oars-1.1"/>
```
We do not have any rating done.  So what does this mean? Seems a bit weird to add.

pemensik replied:
That can be specified at https://hughsie.github.io/oars/

I think it contains unmoderated chat in multiplayer mode, so it should have following spec:

```
 <content_rating type="oars-1.1">
    <content_attribute id="social-chat">intense</content_attribute>
  </content_rating>
```

TrueBrain replied:
We tend to stay away from any rating, as we never did any official rating. So I rather just not mention it at all, than to start filling in a random rating system with the information we think is valid :)

ghisvail replied:
> So I rather just not mention it at all

Flathub requires a rating to pass validation of the AppStream metadata.

> than to start filling in a random rating system with the information we think is valid :)

The specification is versioned, used universally in the Flatpak ecosystem and fits in a [simple webpage](https://hughsie.github.io/oars/generate.html). We can call that many things, but random it is not.

ghisvail replied:
> That can be specified at https://hughsie.github.io/oars/
> 
> I think it contains unmoderated chat in multiplayer mode, so it should have following spec:
> 
> ```
>  <content_rating type="oars-1.1">
>     <content_attribute id="social-chat">intense</content_attribute>
>   </content_rating>
> ```

I have never used OpenTTD online, so I could not verify this. If that's the case, then indeed the OARS metadata need an update.

TrueBrain replied:
So we gathered so far that the rating entry exists so this file can be used to pass validation on Flathub. This isn't clear from the context of this file, but that is fixable.

Sadly, the more pressing issue is that OpenTTD hasn't officially made any comments on how the game is rated. And I feel uncomfortable to do that with a line in a PR that is about a Linux-only file. I guess that should be done more clearly, and more prominent.

Further more, there are many rating systems in the world. We are confronted with most of them when publishing the game on stores. This further complicates the situation, that we have to make sure those ratings are in sync with each other. I guess it was about to happen sooner or later that we need to figure out a few of those rating systems, and put somewhere what the results are. So everyone can align on the same, and avoid exactly what happened in this short thread.

Which means we have two choices for this PR: either we remove the rating, and this PR becomes easier. Or we keep the rating, and we need to figure out (in another PR) how to rate OpenTTD for a few of these rating systems and publish the results somewhere.
This stuff is never easy :(

pemensik replied:
It could be solved by making extra chat plugin, which would not be enabled by default. If some file existed in filesystem, it would enable in-game chat. That way parent-administrator could leave children safe with the game, if they would lack permission to install the extension (remove or install tested file). But I expect children below 13 years are not likely to play this game often. At least compile-time option would be nice for the chat.

ghisvail replied:
>  This stuff is never easy :(

Hence why this [generator](https://hughsie.github.io/oars/generate.html) exists: to save time from upstream developers.

Now, that's a lot of text to say "we don't care". So let's not lose further time on this.

rubidium42 replied:
What is the goal of the rating? How pedantic does one need to be when making the rating?

I would say you forgot moderate money-gambling; for example attempting to bribe local authorities which is practically gambling with virtual money on a successful bribe.
And drugs-alcohol due to FIRS having alcohol as cargo, which is essentially consumed by the human units.
And drugs-tobacco due to Industries of the Caribbean having tobacco as cargo. (https://newgrf-specs.tt-wiki.net/wiki/CargoTypes)
And violence-fantasy as you can deliberately crash trains/vehicles and kill many human units.
And sex-nudity as Long road vehicle 4 has vehicle with imagery that could be recognised as naked women, so there's even a version with those images cut (http://george.zernebok.net/newgrf/downloads.html)

And to be honest... intense desecration is feasible with a NewGRF. Just have an industry that accepts passengers, when passengers arrive they are put in a pen and some animals start feasting on them until they are gone. Does that mean OpenTTD should be rated like that, since it is possible with readily downloadable graphics sets?

In any case, I would rather not self-rate the application as it will open a huge can of worms of people who see things differently. In which the developers are caught in the middle as they cannot please both sides, so if it doesn't need to be provided for anything that the OpenTTD developers provide I rather not add this point of contention.

TrueBrain replied:
> It could be solved by making extra chat plugin, which would not be enabled by default. If some file existed in filesystem, it would enable in-game chat. That way parent-administrator could leave children safe with the game, if they would lack permission to install the extension (remove or install tested file). But I expect children below 13 years are not likely to play this game often. At least compile-time option would be nice for the chat.

I never thought about that, but you make a very good point. And it might surprise you how young our players can be .. at least, some of the older players always pride themselves with telling their N-year-old is playing OpenTTD :P Tnx for this sharp remark, it is a bit of an eye-opener (might sound silly to some, but I am just being honest here).

pemensik replied:
Of course local games are safe for any children and I expect that would be most common. For online play it would be nice, if chat could be hidden on online game. Maybe just messages from the server shown. You can still make some rude images or words using railroads, but I expect that would not allow unwanted chat with the kid. Current unchecked chat allows sexual harassment among other thing unfortunately.

pemensik replied:
> What is the goal of the rating? How pedantic does one need to be when making the rating?

I think this is a best effort thing. What people responsible for the release agree upon.
> 
> I would say you forgot moderate money-gambling; for example attempting to bribe local authorities which is practically gambling with virtual money on a successful bribe. And drugs-alcohol due to FIRS having alcohol as cargo, which is essentially consumed by the human units. And drugs-tobacco due to Industries of the Caribbean having tobacco as cargo. (https://newgrf-specs.tt-wiki.net/wiki/CargoTypes) And violence-fantasy as you can deliberately crash trains/vehicles and kill many human units. And sex-nudity as Long road vehicle 4 has vehicle with imagery that could be recognised as naked women, so there's even a version with those images cut (http://george.zernebok.net/newgrf/downloads.html)

Well, good point. With online content made by 3rd parties and downloaded in the game, it might require extensions to declare how they are rated themselves. I am not aware there is a way to declare account a child one, with age limit, on a Linux desktop. Similar to what Android offers. Might be good to have downloading of new extensions disabled by administrator decision as well.
> 
> And to be honest... intense desecration is feasible with a NewGRF. Just have an industry that accepts passengers, when passengers arrive they are put in a pen and some animals start feasting on them until they are gone. Does that mean OpenTTD should be rated like that, since it is possible with readily downloadable graphics sets?
That would be why even Steam disables extensions in family locked accounts. I would say we should rate what is contained in the game itself, not their extensions. But some future decision might design some concept, how the game should/can protect underage players.
> 
> In any case, I would rather not self-rate the application as it will open a huge can of worms of people who see things differently. In which the developers are caught in the middle as they cannot please both sides, so if it doesn't need to be provided for anything that the OpenTTD developers provide I rather not add this point of contention.

This is not an option already. Flatpak already requires some definition for a reason. I am quite sure Steam does as well, even though there is already OpenTTD without any adult specifications. I would say every current app store has similar thing. Pretending we do not do won't work. Stating explicitly it is only for adults would, but should not be necessary.

pemensik committed:
Remove manual entries of releases

They are inpractical for upstream to maintain.

pemensik committed:
Specify in-game chat is not moderated

It might be inappropriate for underage children in online games. Parents
should be aware of the possibility.

pemensik committed:
Add rule to install metainfo file

Classify it among docs.

pemensik committed:
Change copyright to authors in metainfo

License makes it basically public domain, so it does not make sense to
call anything in it copyright anyway.

Add myself into changes.
